clouddrift/adapters/hurdat2.py

```diff
@@ -0,0 +1,452 @@
+import enum
+import os
+import re
+import tempfile
+from dataclasses import dataclass, field, fields
+from datetime import datetime, timezone
+from io import StringIO
+from uuid import uuid4
+
+import numpy as np
+import xarray as xr
+
+from clouddrift.adapters.utils import download_with_progress
+from clouddrift.raggedarray import RaggedArray
+
+_DEFAULT_NAME = "hurdat2"
+_ATLANTIC_BASIN_URL = "https://www.aoml.noaa.gov/hrd/hurdat/hurdat2.html"
+_NORTHEAST_PACIFIC_BASIN_URL = "https://www.aoml.noaa.gov/hrd/hurdat/hurdat2-nepac.html"
+
+_DEFAULT_FILE_PATH = os.path.join(tempfile.gettempdir(), "clouddrift", _DEFAULT_NAME)
+os.makedirs(_DEFAULT_FILE_PATH, exist_ok=True)
+
+
+class BasinOption(int, enum.Enum):
+    ATLANTIC = 1
+    NORTHEAST_PACIFIC = 2
```

Comment:
Not sure about the `NORTHEAST_PACIFIC` name. One place they call the dataset: 
`Northeast and North Central Pacific hurricane database (HURDAT2) 1949-2022`, and just below `East Pacific Basin`. 

Also, loading the pacific dataset, the `ds.basin` variables,
```
In [22]: np.unique(ds.basin.values)
Out[22]: array(['CP', 'EP'], dtype='<U2')
```
takes two values: Central Pacific and Eastern Pacific. So maybe we just renamed this pacifc to `PACIFIC`?

Reply:
Makes sense to me, updated.